test/e2e/priodic_update_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPeriodicUpdate(t *testing.T) {
```

Comment:
Should we also add a test case here when the resource is non-versioned and the `kapp.k14s.io/max-duration` annotation is present but the `update-strategy` is not `always-replace` to validate we are throwing the error in that scenario?

Reply:
added